test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsClientModelValidatorProviderTest.cs

```diff
@@ -91,13 +93,19 @@ public void UnknownValidationAttribute_IsNotAddedAsValidator()
                 stringLocalizerFactory: null);
             var metadata = _metadataProvider.GetMetadataForType(typeof(DummyClassWithDummyValidationAttribute));
 
-            var providerContext = new ClientValidatorProviderContext(metadata);
+            var providerContext = new ClientValidatorProviderContext(metadata, GetValidatorItems(metadata));
 
             // Act
             provider.GetValidators(providerContext);
 
             // Assert
-            Assert.Empty(providerContext.Validators);
+            var validatorItem = Assert.Single(providerContext.Results);
+            Assert.Null(validatorItem.Validator);
+        }
+
+        private IList<ClientValidatorItem> GetValidatorItems(ModelMetadata metadata)
```

Comment:
We should add a new test for the behavior where we 'synthesize' a client validator for `metadata.IsRequired`.